packages/sdk/src/encryption/PublisherKeyExchange.ts

```diff
@@ -67,10 +75,13 @@ export class PublisherKeyExchange {
         })
         eventEmitter.on('messagePublished', (msg) => {
             if (msg.signatureType === SignatureType.ERC_1271) {
-                const address = msg.getPublisherId()
-                if (!this.erc1271ContractAddresses.has(address)) {
-                    logger.debug('Add ERC-1271 publisher', { address })
-                    this.erc1271ContractAddresses.add(address)
+                const publisherId = msg.getPublisherId()
+                if (isEthereumAddressUserId(publisherId)) {
+                    const address = toEthereumAddress(publisherId)
+                    if (!this.erc1271ContractAddresses.has(address)) {
```

Comment:
Could this be problematic from performance viewpoint given that this event is emitted every time a message is published?